benchmarking/runs/priority_algorithm/diversify_gender_min_2/max_keep.py

```diff
@@ -0,0 +1,240 @@
+from typing import Dict, List
+
+import typer
+
+from api.ai.interfaces.algorithm_config import (
+    PriorityAlgorithmConfig,
+    RandomAlgorithmConfig,
+    WeightAlgorithmConfig,
+)
+from api.ai.priority_algorithm.mutations import (
+    mutate_local_max,
+    mutate_random_swap,
+    mutate_local_max_random,
+    mutate_local_max_double_random,
+    mutate_robinhood,
+    mutate_robinhood_holistic,
+)
+from api.models.enums import ScenarioAttribute, Gender, AlgorithmType
+from benchmarking.data.simulated_data.mock_student_provider import (
+    MockStudentProvider,
+    MockStudentProviderSettings,
+)
+from benchmarking.evaluations.graphing.graph_metadata import GraphData, GraphAxisRange
+from benchmarking.evaluations.graphing.line_graph import line_graph
+from benchmarking.evaluations.graphing.line_graph_metadata import LineGraphMetadata
+from benchmarking.evaluations.interfaces import TeamSetMetric
+from benchmarking.evaluations.metrics.average_gini_index import (
+    AverageGiniIndex,
+)
+from benchmarking.evaluations.metrics.maximum_gini_index import MaximumGiniIndex
+from benchmarking.evaluations.metrics.minimum_gini_index import MinimumGiniIndex
+from benchmarking.evaluations.metrics.priority_satisfaction import PrioritySatisfaction
+from benchmarking.evaluations.scenarios.diversify_gender_min_2_female import (
+    DiversifyGenderMin2Female,
+)
+from benchmarking.runs.interfaces import Run
+from benchmarking.simulation.goal_to_priority import goals_to_priorities
+from benchmarking.simulation.insight import Insight
+from benchmarking.simulation.simulation_set import SimulationSet, SimulationSetArtifact
+from benchmarking.simulation.simulation_settings import SimulationSettings
+
+
+class DiversifyGenderMin2MaxKeep(Run):
+    @staticmethod
+    def start(num_trials: int = 10, generate_graphs: bool = False):
+        """
+        Goal: Run diversify gender scenario while varying the maximum keep argument for the priority algorithm.
+        """
+
+        # Defining our changing x-values (in the graph sense)
+        max_keep = list(range(1, 10))
+        ratio_of_female_students = 0.4
+        max_iterate = 300
```

Comment:
these things being held constant are kinda super arbitrary, how did we arrive at them?